docs/handbook/tutorial.rst

```diff
@@ -522,6 +493,42 @@ Reading from a tar archive
     fp = TarIO.TarIO("Tests/images/hopper.tar", "hopper.jpg")
     im = Image.open(fp)
 
+
+Batch processing
+^^^^^^^^^^^^^^^^
+
+Operations can be applied to multiple image files. For example, all PNG images
+in the current directory can be saved as JPEGs at reduced quality.
+
+::
+
+    import os
+    from PIL import Image
+
+
+    def compress_image(source_path, dest_path):
+        with Image.open(source_path) as img:
+            if img.mode != "RGB":
+                img = img.convert("RGB")
+            img.save(dest_path, "JPEG", optimize=True, quality=80)
+
+
+    paths = [path for path in os.listdir(".") if path.endsWith(".png")]
+    for path in paths:
+        compress_image(path, path[:-4] + ".jpg")
+
+Since images can also be opened from a ``Path`` from the ``pathlib`` module,
+the example could be modified to use ``pathlib`` instead of ``os``.
+
+::
+
+    from pathlib import Path
+
+    paths = [path for path in Path.cwd().iterdir() if path.suffix == ".png"]
```

Comment:
And pathlib can also do globs, something like:
```suggestion
    paths = Path(".").glob("*.png")
```

Reply:
Thanks for the suggestion!
In everyday use I find the first implementation a bit better, because one can easily add further conditions, e.g.
`paths = [path for path in Path.cwd().iterdir() if path.suffix == ".png" and "foo" in path.name and "bar" not in path.name]` , but the other one is easier to read and maybe therefore better in the documentation

Reply:
Then I'd probably do something like:

```diff
-paths = [path for path in Path.cwd().iterdir() if path.suffix == ".png" and "foo" in path.name and "bar" not in path.name]
+paths = (path for path in Path(".").glob("*foo*.png") and "bar" not in path.name)
```

Reply:
Your solution is nice, but is quite nested and therefore a bit difficult to read.
The way I normally do it is like this:
```py
for path in Path.cwd().iterdir(): 
    cond = True
    cond &= "foo" in path.name
    cond &= "bar" not in path.name
    cond &= ".png" == path.suffix
    if cond is True:
        file_names.append(path)
```
this is very readable, and when debugging, one can quickly add and remove further conditions by commenting and uncommenting single code lines.